src/proto/types.rs

```diff
@@ -119,8 +158,10 @@ impl CloseCode {
         match self.0.get() {
             1004 | 1005 | 1006 | 1015 => true,
             1000..=4999 => false,
-            // SAFETY: `TryFrom` is the only way to acquire self and it errors for these values
-            0..=999 | 5000..=u16::MAX => unsafe { unreachable_unchecked() },
+            0..=999 | 5000..=u16::MAX => {
+                debug_assert!(false, "unexpected CloseCode");
+                false
+            }
         }
```

Comment:
I'd expect this code to basically optimize down to `matches!(self.0.get(), 1004 | 1005 | 1006 | 1015)`. The extra branch shouldn't be a problem.

Reply:
The code generated is a bit different, but I don't mind this one. I'd wan to keep the comment as a reasoning for the `debug_assert`, though. https://godbolt.org/z/99bbjcjoM

Reply:
Done